pkg/agent/multicast/mcast_controller.go

```diff
@@ -139,52 +152,97 @@ func (c *Controller) checkLastMember(group net.IP) {
 	c.queue.AddAfter(group.String(), igmpMaxResponseTime)
 }
 
-// clearStaleGroups checks the stale groups which have not been updated for mcastGroupTimeout, and then notifies worker
+// clearStaleGroups checks the stale group members which have not been updated for mcastGroupTimeout, and then notifies worker
 // to remove them from groupCache.
 func (c *Controller) clearStaleGroups() {
 	now := time.Now()
 	for _, obj := range c.groupCache.List() {
 		status := obj.(*GroupMemberStatus)
 		status.mutex.RLock()
 		diff := now.Sub(status.lastIGMPReport)
-		status.mutex.RUnlock()
 		if diff > mcastGroupTimeout {
+			// Notify worker to remove the group from groupCache if all its members are not updated before mcastGroupTimeout.
 			c.queue.Add(status.group.String())
+		} else {
+			// Create a "leave" event for a local member if it is not updated before mcastGroupTimeout.
+			for member, lastUpdate := range status.localMembers {
+				if now.Sub(lastUpdate) > mcastGroupTimeout {
+					ifConfig, exists := c.ifaceStore.GetInterfaceByName(member)
+					if !exists {
+						klog.InfoS("Interface is not found in memory cache, create an empty interfaceConfig", "name", member)
+						ifConfig = &interfacestore.InterfaceConfig{
```

Comment:
I do not understand this. If you do not care about the interfaceConfig exists in cache or not, why you even look up it from cache first?

Reply:
Good catch, remove the query from cache, and create a InterfaceConfig with InterfaceName only.